R/remove_constants.R

```diff
@@ -0,0 +1,61 @@
+#' Remove empty rows and columns and constant column
+#'
+#' @param data The input data frame or linelist
+#' @param cutoff The cut-off for empty rows and columns removal. If provided,
+#'    only rows and columns where the percent of missing data is greater than
+#'    this cut-off will removed.
+#'
+#' @return The input dataset without the empty rows and columns and the constant
+#'    columns.
+#' @export
+#'
+#' @examples
+#' data <- readRDS(system.file("extdata", "test_df.RDS", package = "cleanepi"))
+#'
+#' # introduce an empty column
+#' data$empty_column <- NA
+#'
+#' # remove the constant columns, empty rows and columns
+#' dat <- remove_constants(
+#'   data   = data,
+#'   cutoff = 1
+#' )
+#'
+#' # check the report to see what has happened
+#' report <- attr(dat, "report")
+#' summary(report)
+remove_constants <- function(data, cutoff = 1L) {
+  checkmate::assert_number(cutoff, lower = 0.0, upper = 1.0, na.ok = FALSE,
+                           finite = TRUE, null.ok = FALSE)
+  report     <- attr(data, "report")
+  # remove the empty rows and columns
+  dat        <- data %>%
+    janitor::remove_empty(which = c("rows", "cols"), cutoff = cutoff)
```

Comment:
Good point - will look into this later.

For now, I prefer keeping this as if I have to code this, it will be similar (if not the same) as the code in `janitor::remove_empty`.